### PR TITLE
Add additional_no_proxy variable, wire proxy options into install. 

### PR DESCRIPTION
**Pull request by bnferguson**
## Background

*This code requires an updated version of the `ptfe` tool that's currently unreleased. And as such is not ready to merge.*

We've seen issues with customers that use proxies having issues with cluster creation depending their network setups. One possible problem for this is that we don't currently allow for adding to the `no_proxy` list at install time. 

Replicated does autodetect `http_proxy` (though I've made it explicit here) but it does not detect `no_proxy` other than the defaults it sets itself (for its own communication). So this PR (along with a PR that was just merged in the private `ptfe` repo) is wiring that in. 

~However this is still in WIP because we're seeing some issues here and there around rejoining of nodes and sometimes primary nodes not showing up as primary (this may be an unrelated issue to the proxy). Additionally we may want to make sure we're adding the internal load balancer we use for the cluster API to the no-proxy by default list as I've seen this pop up in proxy logs.~

WIP removed as the primary node rejoining is expected, currently the cluster isn't healthy when a primary goes missing so it won't allow new ones to join. These nodes will expire over an hour or by removing them as outlined here: https://help.replicated.com/community/t/removing-nodes-from-kubernetes-clusters/371 

Also I've added the lb to the no_proxy list. 

## How Has This Been Tested

I've been spinning up clusters with a squid proxy, once they come up properly, I usually delete a primary and re-plan/apply to see if the new node joins the cluster. 

Additionally I watch the `access.log` on the squid proxy to see what requests are coming in and if they should be there (internal calls vs calls to docker/Replicated/GitHub). 

### Test Configuration

* Terraform Version: 0.12.18

## This PR makes me feel

proxy. proxy. proxy.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -206,6 +206,12 @@ variable "http_proxy_url" {
   default     = ""
 }
 
+variable "additional_no_proxy" {
+  type        = string
```
Comment (erindatkinson) [inline variables.tf]: would it make more sense to do this as a list type and use [a join function](https://www.terraform.io/docs/configuration/functions/join.html) in the template file usage?

* [L35](https://github.com/hashicorp/terraform-aws-terraform-enterprise/pull/56/files#diff-264131a8410d0f8f58af1c713d120c53R35)
* [L65](https://github.com/hashicorp/terraform-aws-terraform-enterprise/pull/56/files#diff-264131a8410d0f8f58af1c713d120c53R65)
* [L101](https://github.com/hashicorp/terraform-aws-terraform-enterprise/pull/56/files#diff-264131a8410d0f8f58af1c713d120c53R101)

**Comment by erindatkinson** [comment]
so for https://github.com/hashicorp/terraform-aws-terraform-enterprise/pull/56/files#diff-34153f17d943c3c52e79e94ef931228cR90-R125

rather than having three different /etc/profile.d/proxy.sh writes, what if we wrote something like this

```yaml

- path: /etc/profile.d/proxy.sh
  owner: root:root
  permissions: "0755"
  content: |
    export http_proxy="${proxy_url}"
    export https_proxy="${proxy_url}"
    export no_proxy=10.0.0.0/8,127.0.0.1,169.254.169.254,${cluster_api_lb}
%{ if additional_no_proxy != "" }
    export no_proxy="${no_proxy},${additional_no_proxy}
%{ endif }
%{ if repl_cidr != "" }
    export no_proxy="${no_proxy},${repl_cidr}"
%{ endif }
```

⚠️ not sure where we'd want the `~`'s for line deletion.

File: files/install-ptfe.sh [hunk 2]
```diff
@@ -6,14 +6,24 @@ set -e -u -o pipefail
 if [ -s /etc/ptfe/proxy-url ]; then
   http_proxy=$(cat /etc/ptfe/proxy-url)
   https_proxy=$(cat /etc/ptfe/proxy-url)
+  api_load_balancer_without_port=$(cat /etc/ptfe/cluster-api-endpoint | awk -F ":" '/1/ {print $1}')
```
Comment (erindatkinson) [inline files/install-ptfe.sh]: love this variable name! 💜

**Comment by bnferguson** [comment]
> rather than having three different /etc/profile.d/proxy.sh writes, what if we wrote something like this for readability

Thank you for this review! And yes I totally agree with this. This was done somewhat as a field edit with a customer and this felt so messy at the time. Cleaning this up makes things feel much better than relying on if statements lining up in a certain way to drop a file. That feels very fragile.

File: variables.tf (hunk 1 above)
Comment (bnferguson) [reply variables.tf]: Hmmm, yeah I think so! Especially since wrong formatting of the list can actually mess up some implementations of `no-proxy`.

Comment (bnferguson) [reply variables.tf]: That is to say: good looking out! Thank you!

**Pull request merged**